src/viewport.js

```diff
@@ -1128,6 +1125,26 @@ $.Viewport.prototype = {
         return changed;
     },
 
+    // private
+    _rotateAboutPivot: function(immediately, degrees){
+        //immediately defaults to false; degrees are ignored for immediately==false
```

Comment:
How about we just pass in `degrees` and then test if it's `=== undefined` instead of the `immediately`? Just a thought. 

What caught my eye is that usually `immediately` is the last argument in a function. Of course, this is a private function, so its API doesn't matter too much. 

If you do update the arguments, don't forget to update the comment to make it clear what the difference is (and why).

Reply:
My first implementation was `function(degrees, immediately)` with the same behavior as now, which is that unless `immediately` is truthy, `degrees` is ignored.  That felt a little weird, because calling it as `_rotateAboutPivot(90)` would just totally ignore the parameter and there wasn't an explicit reason to expect that behavior. Since the immediate/not immediate distinction actually drives the branching logic, I thought putting that first might help. But it is still strange, I agree. Leaving `degrees` undefined to mean `do this immediately` also seems not totally clear...

Perhaps a single parameter `immediateDegrees`, and check `immediateDegrees !== undefined` within the function?

Part of the reason this feels odd, I think, is that this while this function does centralize the overall logic into one place, it has to pull in extra stuff from the calling functions (through the arguments) because the difference in immediate vs spring-based happens in the middle of the process.

Reply:
Yeah, it is a little funky having the logic split in the function. I think it's worth it to keep the overall technique centralized, though. I like your `immediateDegrees` idea. And, of course, some good comments to make it clear what's going on :)